tools/configure_file.bzl

```diff
@@ -10,7 +10,7 @@
 
 CMAKE_FALSE_CONSTANTS = ["0", "OFF", "NO", "FALSE", "N", "IGNORE", "NOTFOUND"]
 
-def configure_file_impl(ctx, vars):
+def configure_file_impl(ctx, vars, out):
```

Comment:
Can we do this without breaking the API?

It's going to be a pain to get these changes in as we'll have to merge this update from the top down before we can merge the PR's that implement this.

I think this can be avoided if we just add a `out` attr to the doxygen rule so this can continue to be referenced as `ctx.attr.out`

Reply:
But then we would have to pass the `out` parameter to the `swift_doxygen` rule (because we can't overwrite an attribute inside a rule), which I think might complicate the doxygen rule API. (The `out` parameter is used to generate the Doxyfile "template", not the doxygen output directory)

How about introducing None as a default parameter and if any parameter is not set, taking it from the ctx:
```
def configure_file_impl(ctx, vars = None, out = None):
    if vars == None:
        vars = ctx.attr.vars

    if out == None:
        out = ctx.attr.out
```

Reply:
Yeah that works 👍🏼